prow/cmd/checkconfig/main_test.go

```diff
@@ -357,3 +360,80 @@ func TestOrgRepoUnion(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateUnknownFields(t *testing.T) {
+	testCases := []struct {
+		name        string
+		configBytes []byte
+		config      interface{}
+		expectedErr error
+	}{
+		{
+			name: "valid config",
+			configBytes: []byte(`plugins:
+  kube/kube:
+  - size
+  - config-updater
+config_updater:
+  maps:
+    # Update the plugins configmap whenever plugins.yaml changes
+    kube/plugins.yaml:
+      name: plugins
+size:
+  s: 1`),
+			expectedErr: nil,
+		},
+		{
+			name: "invalid top-level property",
+			configBytes: []byte(`plugins:
+  kube/kube:
+  - size
+  - config-updater
+notconfig_updater:
+  maps:
+    # Update the plugins configmap whenever plugins.yaml changes
+    kube/plugins.yaml:
+      name: plugins
+size:
+  s: 1`),
+			expectedErr: fmt.Errorf("unknown fields present: notconfig_updater"),
+		},
+		{
+			name: "invalid second-level property",
+			configBytes: []byte(`plugins:
+  kube/kube:
+  - size
+  - config-updater
+size:
+  xs: 1
+  s: 5`),
+			expectedErr: fmt.Errorf("unknown fields present: size.xs"),
+		},
+		{
+			name: "invalid array element",
```

Comment:
This is an invalid array not an invalid array element. Array element validation doesn't appear to be tested.

Reply:
Sorry I'm not sure I understand the comment. Test case contains invalid array entry: `triggers` is an array of `Trigger` objects and which should have [`repos`](https://github.com/kubernetes/test-infra/blob/master/prow/plugins/config.go#L336) field. One of the entries has `repoz` so it must be invalid entry.

Could you please clarify what test case you had in mind? Thanks.